pkg/rbeconfigsgen/options.go

```diff
@@ -0,0 +1,258 @@
+	OutputManifest string
+	// PlatformParams specify platform specific constraints used to generate a BUILD file with the
+	// toolchain & platform targets in the generated configs. It's recommended to use the defaults
+	// corresponding to the ExecOS.
+	PlatformParams *PlatformToolchainsTemplateParams
+
+	// C++ Config generation options.
+	// GenCPPConfigs determines whether C++ configs are generated.
+	GenCPPConfigs bool
+	// CPPConfigTargets are the Bazel targets that will be used to make Bazel auto-generated C++
+	// configs.
+	// This field can be auto-populated with a default value. See DefaultExecOptions below.
+	CPPConfigTargets []string
+	// CPPConfigRepo is the name of the Bazel external repo (i.e., the repo name without the '@')
+	// whose Bazel build output directory will be used to extract generated C++ config files.
+	// This field can be auto-populated with a default value. See DefaultExecOptions below.
+	CPPConfigRepo string
+	// CppBazelCmd is the Bazel command that'll be executed (build|query) on CppConfigTargets to
+	// generate CppConfigTargets. So if CppConfigTargets is @foo//:blah & CppBazelCmd is build,
+	// this tool will run bazel build @foo//:blah.
+	// This field can be auto-populated with a default value. See DefaultExecOptions below.
+	CppBazelCmd string
+	// CppGenEnv are the environment variables that'll be set when running the Bazel command to
+	// generate C++ configs inside the toolchain container. Only one of CppGenEnv or CppGenEnvJSON
+	// can be specified.
+	CppGenEnv map[string]string
+	// CppGenEnvJSON is a JSON file with environment variables that'll be set when running the Bazel
+	// command to generate C++ configs inside the toolchain container. This field is union'd with
+	// CppGenEnv. Only one of CppGenEnv or CppGenEnvJSON can be specified.
+	CppGenEnvJSON string
+
+	// Java config generation options.
+	// GenJavaConfigs determines whether Java configs are generated.
+	GenJavaConfigs bool
+	// TempWorkDir is a temporary directory that will be used by this tool to store intermediate
+	// files. If unspecified, a temporary directory will be requested from the OS.
+	TempWorkDir string
+	// Cleanup determines whether the running container & intermediate files will be deleted once
+	// config generation is done. Setting it to false is useful for debugging intermediate state.
+	Cleanup bool
+}
+
+// DefaultOptions are some option values that are populated as default values for certain fields
+// of "Options". See "Options" for explanation on what the fields mean.
+type DefaultOptions struct {
+	PlatformParams   PlatformToolchainsTemplateParams
+	CPPConfigTargets []string
+	CPPConfigRepo    string
+	CppBazelCmd      string
+	CppGenEnv        map[string]string
+}
+
+const (
+	// OSLinux represents Linux when selecting platforms.
+	OSLinux = "linux"
+	// OSWindows represents Windows when selecting platforms.
+	OSWindows = "windows"
+)
+
+var (
+	validOS = []string{
+		OSLinux,
+		OSWindows,
+	}
```

Comment:
I'm going to stick with the string list because:
1. I don't expect to support more than 3-4 OS so the O(n) shouldn't be a concern and is much simpler. Usually for very small lists, an array performs much better (caches, branch prediction, etc) compared to hashes.
2. Given the key join becomes more complicated, I don't see a huge value in refactoring to get rid of the strListContains